### ISSUE TITLE
Custom Table Column cannot be searched and sorted

### ISSUE THREAD

**Issue by icangku** [njleonzhang/vue-data-tables#190]
Please follow the issue template, or your issue may be closed automatically.

**For bug report, provide the following section**

# Online reproduce
` <el-table-column v-for="title in titles" :prop="title.prop" :label="title.label" :key="title.prop" sortable></el-table-column>
        <el-table-column label="Due Date" sortable>
          <template slot-scope="scope" :prop="scope.row.due_date">
            <h5>{{scope.row.due_date | dateName }}</h5>
          </template>
        </el-table-column>`

I'm using "vue-data-tables": "^3.2.0" & vuejs 2 @njleonzhang

# Expected Behavior
I hope custom column can be searched & sorted

# Current Behavior
The custom column cannot be sorted & searched

# Steps to Reproduce

# Detailed Description

**For feature request, provide the following section**
# Motivation / Use Case

# Expected Behavior

# Other Information

**Comment by icangku** [comment]
This is actually element-ui issue, how to solve this is by using prop in every el-table-colum
ex :
` <el-table-column label="Due Date" prop="due_date" sortable>
          <template slot-scope="scope" >
            <h5>{{scope.row.due_date | dateName }}</h5>
          </template>
        </el-table-column>`

**Comment by icangku** [comment]
It's still not working while searching something, but while sorting its working, can anyone help me? @njleonzhang

**Comment by icangku** [comment]
Here's my code, using vue-data-tables 3.2.0 and vuejs 2
   `titles : [{prop: 'groups.name', label : 'Grup'}, {prop: 'departments.name', label : 'Departemen'}, {prop: 
    'categories.name', label : 'Kategori'}], `

     <data-tables :data="data" :actions-def="actionsDef" :search-def='searchDef'>

      searchDef: {
        props: 'groups.name',
        placeholder : 'Search..'
      },

**Comment by icangku** [comment]
but its showing correctly, 
![image](https://user-images.githubusercontent.com/14269158/45273788-6aa90000-b4de-11e8-863c-8a7f231dfe35.png)
Then how to show something like that?

**Comment by icangku** [comment]
help me man

**Comment by njleonzhang** [comment]
@icangku can you provide a online demo？

**Comment by icangku** [comment]
https://github.com/ElemeFE/element/issues/3308#issuecomment-384943011 - gives me the answer, but the question is how to change the default search method

**Comment by njleonzhang** [comment]
I am still confusing on what you want. I think you want to sort the table with some special logic. 
in current version, [customize-sort-function](https://www.njleonzhang.com/vue-data-tables/#/en-us/sort?id=customize-sort-function-for-data-tables), should be able to help you. This feature should also be able to work in 3.2.0. 

please check the [v3.2.0 code](https://github.com/njleonzhang/vue-data-tables/blob/v3.2.0/src/components/DataTables.vue), if you have any question.

**Comment by icangku** [comment]
Yeah, as you said earlier that `groups.name` is not supported so I think I need special logic. So that's why mentioned the link that helps me thank you for answering!

**Comment by icangku** [comment]
> [ElemeFE/element#3308 (comment)](https://github.com/ElemeFE/element/issues/3308#issuecomment-384943011) - gives me the answer, but now the question is how to change the default search method, your documentation using the newest version, but i'm using "vue-data-tables": "^3.2.0",

the function only working for the current not working for all data, but client needs the searching works for all the data, can help me?

**Comment by njleonzhang** [comment]
search also support custom function. please read the doc https://github.com/njleonzhang/vue-data-tables/blob/v3.2.0/docs/searchBoxFilter.md

**Comment by icangku** [comment]
thanks ma man
